werkzeug/wrappers.py

```diff
@@ -990,7 +990,7 @@ def set_data(self, value):
         else:
             value = bytes(value)
         self.response = [value]
-        if self.automatically_set_content_length:
+        if self.automatically_set_content_length and self.status_code != 204:
```

Comment:
True, this change doesn't affect the returned headers. The purpose of this change is to keep the `resp.content_length` header_property consistent with the actual response headers.

Without this change, `resp.content_length` will return `0` even though the actual `Content-Length` will be absent, not `0`.

It seems misleading to have `resp.content_length == 0` in this case instead of `resp.content_length == None`.